File: ServerNjs/controllers/postController.js
/**
* Created by leo@GDGMendoza on 19/09/2015.
*/
exports.createPost = createPost;
exports.getPosts = getPosts;
exports.getPost = getPost;
exports.deletePost = deletePost;
exports.updatePost = updatePost;

function createPost(req, res, next) {
  var post = require('./../models/post.js');
  var PostTitle = req.body.PostTitle;
  var PostText = req.body.PostText;
  var PostCategory = req.body.PostCategory;
  post.createPost(
      PostTitle,
      PostCategory,
      PostText,
      function(response){
          res.setHeader('Content-Type', 'application/json');
          res.send(JSON.stringify(response));
      }
  );
}

function getPosts(req, res, next) {
  var post = require('./../models/post.js');
  var Limit = req.body.Limit;
  var From = req.body.From;
  post.getPosts(
      Limit,
      From,
      function(response){
          res.setHeader('Content-Type', 'application/json');
          res.send(JSON.stringify(response));
      }
  );
}

function getPost(req, res, next) {
  var post = require('./../models/post.js');
  var PostID = req.params.PostID;
  post.getPost(
      PostID,
      function(response){
          res.setHeader('Content-Type', 'application/json');
          res.send(JSON.stringify(response));
      }
  );
}

function deletePost(req, res, next) {
  var post = require('./../models/post.js');
  var PostID = req.params.PostID;
  post.deletePost(
      PostID,
      function(response){
          res.setHeader('Content-Type', 'application/json');
          res.send(JSON.stringify(response));
      }
  );
}

function updatePost(req, res, next) {
  var post = require('./../models/post.js');
  var PostID = req.params.PostID;
  var PostTitle = req.body.PostTitle;
  var PostText = req.body.PostText;
  var PostCategory = req.body.PostCategory;
  post.updatePost(
      PostID,
      PostTitle,
      PostCategory,
      PostText,
      function(response){
          res.setHeader('Content-Type', 'application/json');
          res.send(JSON.stringify(response));
      }
  );
}

File: ServerAsp/Server/Controllers/PostController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Server.Models;

namespace Server.Controllers
{
    //[Route("post")]
    [RoutePrefix("post")]
    public class PostController : Controller
    {
        private Post post = new Post();

        // GET: Post
        //public ActionResult Index()
        //{
        //    return View();
        //}

        // Posibilidad de marcar post activo / inactivo?

        [Route]
        [HttpGet]
        public JsonResult getall(int Limit = 0, int From = 0)
        {
            /*if (postID != null)
            {

            }*/
            List<Post> posts = new List<Post>();
            posts = post.getPosts(Limit,From);
            return Json(new { posts = posts }, JsonRequestBehavior.AllowGet);
        }

        [Route("{PostID:int}")]
        [HttpGet]
        public JsonResult get(int PostID)
        {
            Post result = new Post();
            result = post.getPost(PostID);
            return Json(new { post = result }, JsonRequestBehavior.AllowGet);
        }

        [Route("{PostID:int}")]
        [HttpDelete]
        public JsonResult delete(int PostID)
        {
            string status = string.Empty;
            status = post.deletePost(PostID);
            return Json(new { status = status }, JsonRequestBehavior.AllowGet);
        }

        [Route]
        [HttpPost]
        public JsonResult create(string PostTitle, string PostText)
        {
            string status = string.Empty;
            status = post.createPost(PostTitle, PostText);
            return Json(new { status = status }, JsonRequestBehavior.AllowGet);
        }

        [Route("{PostID:int}")]
        [HttpPut]
        public JsonResult update(int PostID, string PostTitle, string PostText)
        {
            string status = string.Empty;
            status = post.updatePost(PostID, PostTitle, PostText);
            return Json(new { status = status }, JsonRequestBehavior.AllowGet);
        }

        [Route("testConnection")]
        [HttpPost]
        public JsonResult testConnection()
        {
            return Json(new { connection = "ok" }, JsonRequestBehavior.AllowGet);
        }
    }
}